src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.iOS.cs

```diff
@@ -164,13 +164,17 @@ private void AddItems(int firstItem, int count, int section)
 			NativePanel?.InsertItems(GetIndexPathsFromStartAndCount(firstItem, count, section));
 
 			ManagedVirtualizingPanel?.GetLayouter().AddItems(firstItem, count, section);
+
+			Refresh();
```

Comment:
Calling Refresh() every time an item is added or removed is a very blunt instrument, this will hurt performance too much in the general case since it will rebuild the state of the list each time. A more targeted fix needs to be found for the linked bugs.

In specific usage contexts where the performance compromise is acceptable, there's actually a flag which enables exactly this behavior, `RefreshOnCollectionChanged`:

https://github.com/unoplatform/uno/blob/b2957200fbc057e1c31caf483bee716f215c8871/src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs#L35-L39

Enabling this flag may be interesting if you're blocked by this bug.

Reply:
Yeah, I agree wholeheartedly with your observation. Using that flag might be the life-saver i'm looking for. Thanks!

Reply:
@davidjohnoliver How would I set this flag? I can't seem to see it on the XAML or from the code behind

Reply:
@davidjohnoliver How would I set this flag? I can't seem to see it on the XAML or from the code behind

Reply:
You should be able to set it directly on the `ListView` in XAML, with an [`ios` or `xamarin` prefix](https://platform.uno/docs/articles/platform-specific-xaml.html?q=conditional#available-prefixes) since it's a property that doesn't exist in UWP.